core/adapters/eth_tx_test.go

```diff
@@ -537,3 +536,157 @@ func TestEthTxAdapter_Perform_NotConnected(t *testing.T) {
+		func(_ interface{}, data ...interface{}) error {
+			secondTxData = data
+			return nil
+		})
+	receipt := models.TxReceipt{Hash: hash, BlockNumber: cltest.Int(confirmed)}
+	ethMock.Register("eth_getTransactionReceipt", receipt)
+	ethMock.Register("eth_blockNumber", utils.Uint64ToHex(safe))
+
+	data = adapter.Perform(input, store)
+	assert.NoError(t, data.GetError())
+
+	// The first and second transaction should have the same data
+	assert.Equal(t, firstTxData, secondTxData)
+
+	// Only one transaction should be created
+	from := cltest.GetAccountAddress(store)
+	txs, err := store.TxFrom(from)
+	require.NoError(t, err)
+	require.Len(t, txs, 1)
+	assert.Len(t, txs[0].Attempts, 1)
+
+	ethMock.EventuallyAllCalled(t)
+}
+
+func TestEthTxAdapter_Perform_NoDoubleSpendOnSendTransactionFailAndNonceChange(t *testing.T) {
+	t.Parallel()
+
+	app, cleanup := cltest.NewApplicationWithKey()
+	defer cleanup()
+	store := app.Store
+	config := store.Config
+
+	ethMock := app.MockEthClient(cltest.Strict)
+	ethMock.Register("eth_getTransactionCount", `0x1`)
+	ethMock.Register("eth_getTransactionCount", `0x2`)
+	app.AddUnlockedKey()
+
+	address := cltest.NewAddress()
+	fHash := models.HexToFunctionSelector("b3f98adc")
+	dataPrefix := hexutil.Bytes(
+		hexutil.MustDecode("0x0000000000000000000000000000000000000000000000000045746736453745"))
+	inputValue := "0x9786856756"
+
+	assert.Nil(t, app.StartAndConnect())
+
+	// Run the adapter, but make sure the transaction sending fails
+
+	hash := cltest.NewHash()
+	sentAt := uint64(9183)
+	ethMock.Register("eth_getBalance", "0x100")
+	ethMock.Register("eth_call", "0x100")
+	ethMock.Register("eth_blockNumber", utils.Uint64ToHex(sentAt))
+	var firstTxData []interface{}
+	ethMock.Register("eth_sendRawTransaction", hash,
+		func(_ interface{}, data ...interface{}) error {
+			firstTxData = data
+			return errors.New("no bueno")
+		})
+	ethMock.Register("eth_blockNumber", utils.Uint64ToHex(sentAt))
+
+	adapter := adapters.EthTx{
+		Address:          address,
+		DataPrefix:       dataPrefix,
+		FunctionSelector: fHash,
+	}
+	input := cltest.RunResultWithResult(inputValue)
+	input.CachedJobRunID = uuid.Must(uuid.NewV4()).String()
+	data := adapter.Perform(input, store)
+	assert.Error(t, data.GetError())
+
+	// Run the adapter again
+
+	confirmed := sentAt + 1
+	safe := confirmed + config.MinOutgoingConfirmations()
+	ethMock.Register("eth_blockNumber", utils.Uint64ToHex(confirmed))
+
+	var secondTxData []interface{}
+	ethMock.Register("eth_sendRawTransaction", hash,
+		func(_ interface{}, data ...interface{}) error {
+			secondTxData = data
+			return nil
+		})
+	receipt := models.TxReceipt{Hash: hash, BlockNumber: cltest.Int(confirmed)}
+	ethMock.Register("eth_getTransactionReceipt", receipt)
+	ethMock.Register("eth_blockNumber", utils.Uint64ToHex(safe))
+
+	data = adapter.Perform(input, store)
+	assert.NoError(t, data.GetError())
+
+	// Since the nonce (and from address) changed, the data should also change
```

Comment:
If either the account or the nonce change, an additional transaction should be created. Only if the gas changes should you not create a new transaction, and instead create a new transaction attempt. Seems like this deviates from that, why?

Reply:
Another account is added: https://github.com/smartcontractkit/chainlink/pull/1239/files/3733b5a2524e6e24afb66e2d779ef0967bd9d579#diff-0d44f8ae519e9f175ac946337ffb211bR627